src/main/java/com/suchocki/parkingmeter/service/DriverPaymentServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.suchocki.parkingmeter.service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.suchocki.parkingmeter.dao.DriverPaymentDAO;
+import com.suchocki.parkingmeter.entity.Currency;
+import com.suchocki.parkingmeter.entity.DriverCharge;
+import com.suchocki.parkingmeter.entity.DriverPayment;
+
+@Service
+public class DriverPaymentServiceImpl implements DriverPaymentService {
+
+	@Autowired
+	private DriverPaymentDAO driverPaymentDAO;
+
+	@Autowired
+	private CurrencyService currencyService;
+
+	@Override
+	public void save(DriverPayment driverPayment) {
+		driverPaymentDAO.save(driverPayment);
+	}
+
+	@Override
+	public DriverPayment get(int id) {
+		return driverPaymentDAO.get(id);
+	}
+
+	@Override
+	public List<DriverPayment> getAll() {
+		return driverPaymentDAO.getAll();
+	}
+
+	@Override
+	public List<DriverPayment> getByDay(Date date) {
+		return driverPaymentDAO.getByDay(date);
+	}
+
+	@Override
+	public List<DriverCharge> getPaymentsSumByday(Date date) {
+
+		Map<Currency, BigDecimal> paymentsSums = new LinkedHashMap<>(); // payments sums in different currencies
+		List<Currency> currencies = currencyService.getAll();
+
+		if (currencies == null) {
+			return null;
```

Comment:
Jak masz zwrócić listę i zwracasz nulla to to jest gruby problem

Reply:
Rzeczywiście, lepiej byłoby zwrócić pustą lub "rzucić" wyjątkiem.